src/client/action/notifications.ts

```diff
@@ -1,6 +1,10 @@
 import { MatrixClient } from "matrix-js-sdk";
 
 export async function markAsRead(mx: MatrixClient, roomId: string) {
+  // Won't mark messages (or whatever in room) as read when tab inactive
+  // or when browser is inactive (i.e. user interacts with some another app)
+  if (document.hidden || !document.hasFocus()) return;
+
```

Comment:
It seems like a wrong place to add such conditioning as this function intent is to mark room as read. A good place for require changes would be the code where `markAsRead` is called (i guess in `useEventArrive` hook in `RoomTimeline.tsx` file)

Reply:
I guess you are right. Thanks for suggestions, I'll rewrite my code when I'll have enough time!